test/types/lib.spec.ts

```diff
@@ -49,28 +53,28 @@ const main = async () => {
     ],
     custom_claim: "CustomValue"
   });
-  expectNotAssignable<OktaJwtVerifier.JwtClaims>({
+  expect<OktaJwtVerifier.JwtClaims>().type.not.toMatch({
     exp: 'not-a-number'
   });
+});
 
-  // JWT Header
-  expectAssignable<OktaJwtVerifier.JwtHeader>({
-    alg: 'RS256',
+test("JWT Header", () => {
+  expect<OktaJwtVerifier.JwtHeader>().type.toBeAssignable({
+    alg: 'RS256' as const,
     kid: "45js03w0djwedsw",
     typ: 'JWT'
   });
-  expectNotAssignable<OktaJwtVerifier.JwtHeader>({
-    alg: 'unsupported-alg',
+  expect<OktaJwtVerifier.JwtHeader>().type.not.toMatch({
+    alg: 'unsupported-alg' as const,
```

Comment:
The target type is missing required `typ` property, hence `expectNotAssignable` is not necessary failing because of `alg` property type mismatch.

Sounds like better idea to use `.toMatch()` here, because it will compare only types of `alg` property.